workflow/source.go

```diff
@@ -138,6 +148,37 @@ func (d Document) IsRemote() bool {
 	return getFileStatus(d.Location) == fileStatusRemote
 }
 
+func (d Document) IsSpeakeasyRegistry() bool {
+	return strings.Contains(d.Location, "registry.speakeasyapi.dev")
+}
+
+func (d Document) ParseSpeakeasyRegistryReference() *SpeakeasyRegistryDocument {
+	if !d.IsSpeakeasyRegistry() {
+		return nil
+	}
+
+	registryDocument := &SpeakeasyRegistryDocument{}
+	subPath := strings.Split(d.Location, "registry.speakeasyapi.dev/")[1]
+	components := strings.Split(subPath, "/")
```

Comment:
is it possible to use `url.Parse` instead of direct string manipulation?

Reply:
I'll look at it